OrbitGl/App.cpp

```diff
@@ -111,6 +111,52 @@ void OrbitApp::SetCommandLineArguments(const std::vector<std::string>& a_Args) {
   }
 }
 
+static bool IsTrue(bool* flag) { return *flag; }
+
+void OrbitApp::OnCaptureStarted() {
```

Comment:
So this is *not* scheduled on the main thread and has to wait for the scheduled action to complete on the main thread, right?

Reply:
right.

Reply:
This I don't understand, the capture start is initiated from the UI, so on the main thread, shouldn't OnCaptureStarted() already be on the main thread? 

In OrbitApp::StartCapture(), on the main thread, we schedule the call to capture_client_->Capture() on a worker thread. We then in turn call "OnCaptureStarted" from the worker thread which needs to schedule some work on the main thread. Are those hops necessary? Why can't we call capture_client_->Capture() from the main thread?

Reply:
So the end-goal is to move state from Capture to CaptureClient, in order for this to be consitent it will need to be protected by mutex and we want to avoid making the mutex and implementation details for synchronization global. This is why I added CaptureStarted/Complete as part of Capture lifecycle. Note that this will also improve error handling, - for example in the case of unsuccessful capture you do not end up cleaning up previous results.

We do not call anything that involves blocking I/O on the main thread in order to avoid freezes to UI. CaptureClient::Capture calls streaming grpc Read which could block until next event has arrived.

Reply:
Ah yes, thanks for the precision.